edenia/antelope-tools
# Fix/ block history and add historical utilization of net and cpu

### Fix block history performance and add net and cpu utilization per block

### What does this PR do?

- Improve the performance of saving block data
  - Redesign block_history table to use less disk space and improve insertion
  - Remove unused indexes in block_history table
  - Insert many block at once instead of one at a time
  - Momentarily request to the ws one message at a time instead of a thousand
- When the connection with the socket fails it waits a minute and try again
- Reduce the frecuency when the last transactions per hour, week and day is updated
- Remove transaction_history field from stats table
- Use the average of transaction per block in the transactions query

### Steps to test

1. Run the project locally with the state history plugin in libre blockchain
2. Check the block_history table
3. Go to block distribution page and check the results
4. Go to Stress test dashboard page and check historical block history
5. Check that the other pages such as block producers, nodes, cpu benchmark are working
